### PR TITLE
Fix data sync

### PR DESCRIPTION
**Pull request by fnh-ssb**

### CODE REVIEW

**Comment by fnh-ssb** [comment]
the actual behaviours is that since the primary key for dataset is path and timestamp, the up to date dataset is set to 1 (clean) and older version of data is set to 0 (dirty).

**Comment by fnh-ssb** [comment]
how to observe: after the data-access is ready, you can execute ./bin/run-scenarios.sh exec local skatt-test and use the load-aggregate-save.ipynb case.
execute the section below:
%%time
konto.write\
    .option("valuation", "INTERNAL")\
    .option("state", "INPUT")\
    .path("/felles/skatt/konto-flatt")

to save data several times to see the catalog database records update.

File: src/main/java/no/ssb/dapla/catalog/dataset/CatalogGrpcService.java [hunk 1]
```diff
@@ -13,16 +13,7 @@
...
-import no.ssb.dapla.catalog.protobuf.Dataset;
-import no.ssb.dapla.catalog.protobuf.DeleteDatasetRequest;
-import no.ssb.dapla.catalog.protobuf.DeleteDatasetResponse;
-import no.ssb.dapla.catalog.protobuf.GetDatasetRequest;
-import no.ssb.dapla.catalog.protobuf.GetDatasetResponse;
-import no.ssb.dapla.catalog.protobuf.ListByPrefixRequest;
-import no.ssb.dapla.catalog.protobuf.ListByPrefixResponse;
-import no.ssb.dapla.catalog.protobuf.SaveDatasetRequest;
-import no.ssb.dapla.catalog.protobuf.SaveDatasetResponse;
+import no.ssb.dapla.catalog.protobuf.*;
```
Comment (RuneLind) [inline src/main/java/no/ssb/dapla/catalog/dataset/CatalogGrpcService.java:16]: The java module system don't like .* we should specify every import. (You can change this setting in intellij)

File: src/main/java/no/ssb/dapla/catalog/dataset/CatalogGrpcService.java [hunk 2]
```diff
@@ -238,4 +229,39 @@ public void delete(DeleteDatasetRequest request, StreamObserver<DeleteDatasetRes
             }
         }
     }
+
+    @Override
+    public void pollute(PolluteDatasetRequest request, StreamObserver<PolluteDatasetResponse> responseObserver){
```
Comment (RuneLind) [inline src/main/java/no/ssb/dapla/catalog/dataset/CatalogGrpcService.java:234]: Would try to find a more describing name. Se comment in https://github.com/statisticsnorway/dapla-catalog-protobuf/pull/2#pullrequestreview-466691629

File: src/main/java/no/ssb/dapla/catalog/dataset/CatalogHttpService.java [hunk 3]
```diff
@@ -38,6 +38,41 @@ public void update(Routing.Rules rules) {
         LOG.info("rules: {}", rules);
         rules.get("/", this::doGetList);
         rules.get("/{pathPart}", this::doGetList);
+        rules.post("/pollute", this::setDirty);
+
+    }
+
+    private void setDirty(ServerRequest req, ServerResponse res) {
```
Comment (RuneLind) [inline src/main/java/no/ssb/dapla/catalog/dataset/CatalogHttpService.java:45]: Would change name to `setWriteInProgress`

File: src/main/java/no/ssb/dapla/catalog/dataset/DatasetRepository.java [hunk 4]
```diff
@@ -48,11 +48,18 @@ public DatasetRepository(SQLClient client) {
...
+                .rxUpdateWithParams("update dataset set isDirty = ? where path = ?", params)
+                .map(UpdateResult::getUpdated);
+    }
+
     public Flowable<Dataset> listDatasets(String pathPart, int limit) {
         JsonArray params = new JsonArray().add( (pathPart != null && pathPart.length() >0) ? "%" + pathPart + "%" : "%").add(limit);
         return client
                 .rxQueryStreamWithParams("SELECT DISTINCT ON (path) " +
-                        " path, document FROM Dataset WHERE path LIKE ? " +
+                        " path, document, isDirty FROM Dataset WHERE path LIKE ? " +
```
Comment (RuneLind) [inline src/main/java/no/ssb/dapla/catalog/dataset/DatasetRepository.java:62]: maybe rename isDirty to `isWriteInProgress`

File: src/main/java/no/ssb/dapla/catalog/dataset/CatalogGrpcService.java (hunk 1 above)
Comment (fnh-ssb) [reply src/main/java/no/ssb/dapla/catalog/dataset/CatalogGrpcService.java:16]: done

File: src/main/java/no/ssb/dapla/catalog/dataset/DatasetRepository.java (hunk 4 above)
Comment (fnh-ssb) [reply src/main/java/no/ssb/dapla/catalog/dataset/DatasetRepository.java:62]: ok, I can rename the attribute both in grpc protocol and database. maybe `isIntact ` is a better name?

**Comment by fnh-ssb** [comment]
will reopen after implement new design.

**Pull request closed without merging**